### PR TITLE
cares: latching if cares has only the default nameserver

### PR DESCRIPTION
**Pull request by alyssawilk**
SupportsCustomResolversAsFallback has been failing locally for me because there's only a default resolver, so we use the configured failovers immediately. 
Turns out branching on isCaresDefaultTheOnlyNameserver in the test doesn't work, because once we install the failovers, it's no longer the only nameserver.
Latching and using, so I finally have green tests

### CODE REVIEW

File: test/extensions/network/dns_resolver/cares/dns_impl_test.cc [hunk 1]
```diff
@@ -477,7 +478,12 @@ TEST_F(DnsImplConstructor, SupportsCustomResolversAsFallback) {
   int result = ares_get_servers_ports(peer->channel(), &resolvers);
   EXPECT_EQ(result, ARES_SUCCESS);
   EXPECT_EQ(resolvers->family, AF_INET);
-  EXPECT_STRNE(inet_ntop(AF_INET, &resolvers->addr.addr4, addr4str, INET_ADDRSTRLEN), "1.2.3.4");
+  if (peer->isCaresDefaultTheOnlyNameserver()) {
```
Comment (mattklein123) [inline test/extensions/network/dns_resolver/cares/dns_impl_test.cc:481]: Sorry I'm confused. How does this fail for you and not anyone else? What is different about your environment?

Comment (mattklein123) [reply test/extensions/network/dns_resolver/cares/dns_impl_test.cc:481]: Is it that you don't have a working resolve.conf per the above comment? I think I probably just don't have enough context on what this test is testing.

Comment (alyssawilk) [reply test/extensions/network/dns_resolver/cares/dns_impl_test.cc:481]: yeah no working resolve.conf, so only the default resolver, at which point the resolver will always use failovers (even if they're configured to only be failovers)
the test tests that failovers aren't used, and so fails for me.

Comment (mattklein123) [reply test/extensions/network/dns_resolver/cares/dns_impl_test.cc:481]: `isCaresDefaultTheOnlyNameserver` gets called from `initializeChannel` which gets called multiple times throughout the session. Are we sure this doesn't change existing behavior? Either way do you mind adding some more comments in this test to say why we branch on this (what the host environment would need to look at)? I'm worried someone else is going to come along and get confused?

/wait

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**